engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -448,6 +450,14 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      if (UserGroupInformation.isSecurityEnabled() && customKeystoreConfig != null && customKeystorePath != null) {
```

Comment:
While it is unusual to enable SSL without kerberos authentication, it is still allowed and possible. So it is not necessary that isSecurityEnabled will be true.